event_response_tool/layer_loss_duplicator/duplicate_layer_loss.py

```diff
@@ -0,0 +1,348 @@
+        else:
+            alert.error(
+                f"LossSet {loss_set.id} was uploaded, but failed while processing. {loss_set.status_message}"
+            )
+
+        return loss_set
+
+    def modify_layer(self, layer):
+        # Recursively process the layer structure and transform loss sets
+        if hasattr(layer, "ref_id"):
+            layer = analyzere.LayerView.retrieve(layer.ref_id).layer
+        if hasattr(layer, "analysis_profile"):
+            layer = layer.layer
+        if type(layer) in [analyzere.LayerView, Reference]:
+            layer = layer.layer
+        if layer.type == "BackAllocatedLayer":
+            logger.exception("Back Allocated layers are not supported")
+            raise
+
+        if layer.type == "NestedLayer":
+            layer.sink = self.modify_layer(layer.sink)
+            layer.sources = [self.modify_layer(s) for s in layer.sources]
+
+        if hasattr(layer, "loss_sets"):
+            try:
+                # no losses to process
+                if len(layer.loss_sets) == 0 and layer.type != "FilterLayer":
+                    pass
+                # Need to replace Filter Layers with unlimited Generic layer
+                elif layer.type == "FilterLayer":
+                    # Transform any loss sets that the Filter Layer may have
+                    loss_sets = []
+
+                    for loss_set in layer.loss_sets:
+                        transformed = self.transform_loss_set(loss_set)
+                        if transformed:
+                            # Add transformed loss set to cache
+                            self.loss_set_mapping[
+                                loss_set.id
+                            ] = transformed.id
+                            loss_sets.append(transformed)
+
+                    layer = self.replace_filter_layer(loss_sets)
+                else:
+                    # Transform the loss sets in the loss set list in-place
+                    loss_sets = []
+
+                    for index, loss_set in enumerate(layer.loss_sets):
+                        # Skip YELTs and parametric loss sets
+                        if (
+                            loss_set.type == "ParametricLossSet"
+                            or loss_set.type == "YELTLossSet"
+                        ):
+                            alert.warning(
+                                f"Encountered {loss_set.type} {loss_set.id}, skipping transformation"
+                            )
+                            loss_sets.append(loss_set)
+
+                        # Check if loss set is already processed
+                        elif loss_set.id in self.loss_set_mapping:
+                            loss_sets.append(
+                                analyzere.LossSet.retrieve(
+                                    self.loss_set_mapping[loss_set.id]
+                                )
+                            )
+
+                        # If unknown loss set, transform it.
```

Comment:
Instead I guess you could check for ELT type, transform it, and skip everything else?